### ISSUE TITLE
`Communications`: Implement Quick Emoji Reactions via Hover on Messages

### ISSUE THREAD

**Issue by egekurt123** [ls1intum/Artemis#8592]
### Problem
Users are able to react to messages using emojis, which involves clicking on the reaction button and then selecting an emoji from a list. This method, while functional, requires multiple steps and can interrupt the flow of using the application.

### Proposed System 
This issue proposes the enhancement of our current messaging feature by introducing "quick reactions." Currently, users can react to messages with emojis by selecting them from a menu. The goal is to streamline this process by enabling quick reactions when hovering over the reaction part of a message.

## UI/UX Design Prototype by @JohannesWt 
![Screenshot 2024-05-21 at 18 07 29](https://github.com/ls1intum/Artemis/assets/69898283/615bc400-14cc-4307-b698-8e123f71c529)

**Comment by zhngharry** [comment]
Ive looked into it and the currently used emoji-library doesn't have an option for a quick-reaction hover, so we would have to implement the hover component manually. 

Which emojis should be shown, a fixed set of emojis, or the most used emojis of the user?

**Comment by egekurt123** [comment]
I don't think we need personalized emojis here, since our goal is to provide a communication service tailored to education rather than a personal messaging service. "Thumbs up", "tick" and maybe one more useful emoji should be great

**Comment by zhngharry** [comment]
@egekurt123 How far along are you with #8591 ? I suppose there might be merge conflicts if we work on this simultanously

**Comment by egekurt123** [comment]
I just started working on it, should be done in max 2 weeks

**Comment by zhngharry** [comment]
Alright, Ive also started working on it, but I dont think there will be big issues once your changes have been applied. I will just readjust then 😄
